diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3bfd08e..99a0171 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -201,7 +201,7 @@ if(WITH_CUPTI)
           ${CUDAToolkit_TARGET_DIR}/extras/CUPTI/include)
       list(APPEND CALI_CUPTI_INCLUDE_DIRS ${_real_cupti_include_dir})
       list(REMOVE_DUPLICATES CALI_CUPTI_INCLUDE_DIRS)
-      if (BUILD_SHARED_LIBS)
+      if (BUILD_SHARED_LIBS OR NOT TARGET CUDA::cupti_static)
         list(APPEND CALIPER_EXTERNAL_LIBS CUDA::cupti)
       else()
         find_library(RT_LIB rt REQUIRED)
